WooCommerce/src/androidTest/kotlin/com/woocommerce/android/screenshots/products/ProductListScreen.kt

```diff
@@ -1,21 +1,75 @@
 package com.woocommerce.android.screenshots.products
 
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.assertion.ViewAssertions
+import androidx.test.espresso.contrib.RecyclerViewActions
+import androidx.test.espresso.matcher.ViewMatchers
 import com.woocommerce.android.R
-import com.woocommerce.android.screenshots.TabNavComponent
+import com.woocommerce.android.screenshots.util.ProductData
 import com.woocommerce.android.screenshots.util.Screen
+import org.hamcrest.Matchers
 
 class ProductListScreen : Screen {
     companion object {
         const val LIST_VIEW = R.id.productsRecycler
-        const val LIST_ITEM = R.id.linearLayout
     }
 
-    val tabBar = TabNavComponent()
-
     constructor() : super(LIST_VIEW)
 
-    fun selectProduct(index: Int): SingleProductScreen {
-        selectItemAtIndexInRecyclerView(index, LIST_VIEW, LIST_ITEM)
+    fun scrollToProduct(productTitle: String): ProductListScreen {
+        // this just checks scrolling
+        Espresso.onView(ViewMatchers.withId(LIST_VIEW)).perform(
+            RecyclerViewActions.actionOnItem<RecyclerView.ViewHolder>(
+                ViewMatchers.hasDescendant(
+                    ViewMatchers.withText(
+                        productTitle,
+                    )
+                ),
+                ViewActions.scrollTo()
+            )
+        )
+        return this
+    }
+
+    fun assertProductCardMatchesMockData(product: ProductData): ProductListScreen {
```

Comment:
A note on naming for the method. This method accepts ProductData as a parameter. An instance of ProductData can be created from any source: mocks, arrays, etc... but this method will still work, and will work in the same way for all cases. Because of this, I think the `MatchesMockData` part creates a somewhat false impression that this will work with mocks only. In reality, this screen does not care about the data source (which is good!)

```suggestion
    fun assertProductCard(product: ProductData): ProductListScreen {
```

Reply:
Done in 69929cce0679644693fe531c113eac6983f01259